src/Qir/Runtime/lib/QIR/QubitManagerRestrictedReuse.hpp

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft Corporation.
```

Comment:
Up to you: Would be nice to have a description of what you told me, what the qubit manager is for, etc.
Either as a section in the existing `README.md` or in a separate `QubitManager.md`.

Reply:
I added concise description as a comment in the .hpp file.